File: 19.12.25/Roman_to_integer.java
class Solution {
    
    public int romanToInt(String s) {
        
        int result = 0;

        for (int i = 0; i<s.length(); i++){
            
            char roman = s.charAt(i);
            char nextRoman = 'n';

            if (i < s.length()-1) {
                nextRoman = s.charAt(i+1);
            }

            int romanInt = RomanToInteger(roman);
            int nextRomanInt = RomanToInteger(nextRoman);

            if (romanInt >= nextRomanInt){
                result += romanInt;
            }
            else{
                result += nextRomanInt-romanInt;
                i++;
            }    
        }
        return result;
    }

    public int RomanToInteger(char roman){
        
        int romanInt = 0;

        switch (roman){
            case 'I':
                romanInt = 1;
            break;

            case 'V':
                romanInt = 5;
            break;

            case 'X':
                romanInt = 10;
            break;

            case 'L':
                romanInt = 50;
            break;

            case 'C':
                romanInt = 100;
            break;

            case 'D':
                romanInt = 500;
            break;

            case 'M':
                romanInt = 1000;
            break;

            case 'n':
                romanInt = 0;
            break;
        }
        return romanInt;
    }
}

File: 27.12.25/roman_to_integer.py
class Solution(object):
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """

        roman_to_int_dict = {
            'I': 1,
            'V': 5,
            'X': 10,
            'L': 50,
            'C': 100,
            'D': 500,
            'M': 1000
        }

        total = 0
        i = 0
        
        while i < len(s):
            if i < len(s) - 1 and roman_to_int_dict.get(s[i]) < roman_to_int_dict.get(s[i + 1]):
                total -= roman_to_int_dict.get(s[i])
            else:
                total += roman_to_int_dict.get(s[i])

            i += 1

        return total
